### DISCUSSION TITLE
What is the significance of creating `span::Id` to determine if they are the same?

### DISCUSSION THREAD

**Discussion by rain2307** [tokio-rs/tracing#3133] [Q&A] [answered]
What negative impacts could arise if I create an infinitely incrementing ID?

https://github.com/tokio-rs/tracing/blob/91fedc1f45d0d4eaafb46917d57fe0e8b8b467cb/tracing-core/src/collect.rs#L257

**Comment by mladedav** [comment, marked as answer]
If you're writing a collector, you can assign IDs like that if you want to.

The registry in tracing itself doesn't do that because it reuses IDs for performance reasons.
